cmd/siftest/test.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	lp, err := clpQueryClient.GetLiquidityProvider(context.Background(), &clptypes.LiquidityProviderReq{
+		Symbol:    "ceth",
+		LpAddress: key.GetAddress().String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	if !lp.LiquidityProvider.LiquidityProviderUnits.Sub(lpBefore.LiquidityProvider.LiquidityProviderUnits).Equal(lpUnits) {
+		return errors.New(fmt.Sprintf("liquidity provided unit mismatch (expected: %s received: %s)",
+			lpUnits.String(),
+			lp.LiquidityProvider.LiquidityProviderUnits.String()),
+		)
+	}
+
+	return nil
+}
+
+func TestSwap(clientCtx client.Context, txf tx.Factory, key keyring.Info) error {
+
+	bankQueryClient := banktypes.NewQueryClient(clientCtx)
+	cethBefore, err := bankQueryClient.Balance(context.Background(), &banktypes.QueryBalanceRequest{
+		Address: key.GetAddress().String(),
+		Denom:   "ceth",
+	})
+	if err != nil {
+		return err
+	}
+	rowanBefore, err := bankQueryClient.Balance(context.Background(), &banktypes.QueryBalanceRequest{
+		Address: key.GetAddress().String(),
+		Denom:   "rowan",
+	})
+	if err != nil {
+		return err
+	}
+
+	clpQueryClient := clptypes.NewQueryClient(clientCtx)
+	poolBefore, err := clpQueryClient.GetPool(context.Background(), &clptypes.PoolReq{Symbol: "ceth"})
+	if err != nil {
+		return err
+	}
+
+	msg := clptypes.MsgSwap{
+		Signer:             key.GetAddress().String(),
+		SentAsset:          &clptypes.Asset{Symbol: "ceth"},
+		ReceivedAsset:      &clptypes.Asset{Symbol: "rowan"},
+		SentAmount:         sdk.NewUint(10000),
+		MinReceivingAmount: sdk.NewUint(0),
+	}
+
+	if _, err := buildAndBroadcast(clientCtx, txf, key, &msg); err != nil {
+		return err
+	}
+
+	cethAfter, err := bankQueryClient.Balance(context.Background(), &banktypes.QueryBalanceRequest{
+		Address: key.GetAddress().String(),
+		Denom:   "ceth",
+	})
+	if err != nil {
+		return err
+	}
+	rowanAfter, err := bankQueryClient.Balance(context.Background(), &banktypes.QueryBalanceRequest{
+		Address: key.GetAddress().String(),
+		Denom:   "rowan",
+	})
+	if err != nil {
+		return err
+	}
+	poolAfter, err := clpQueryClient.GetPool(context.Background(), &clptypes.PoolReq{Symbol: "ceth"})
+	if err != nil {
+		return err
+	}
+
+	rowanDiff := rowanAfter.Balance.Amount.Sub(rowanBefore.Balance.Amount)
+	// negative
+	cethDiff := cethAfter.Balance.Amount.Sub(cethBefore.Balance.Amount)
+	// negative
+	poolNativeDiff := poolBefore.Pool.NativeAssetBalance.Sub(poolAfter.Pool.NativeAssetBalance)
+	poolExternalDiff := poolAfter.Pool.ExternalAssetBalance.Sub(poolBefore.Pool.ExternalAssetBalance)
+
+	fmt.Printf("Pool sent diff: %s\n", poolNativeDiff.String())
+	fmt.Printf("Pool received diff: %s\n", poolExternalDiff.String())
+	fmt.Printf("Address received diff: %s\n", rowanDiff.String())
+	fmt.Printf("Address sent diff: %s\n", cethDiff.String())
+
+	return nil
+}
+
+func TestOpenPosition(clientCtx client.Context, txf tx.Factory, key keyring.Info) error {
```

Comment:
Naming convention question: why some functions are prefixed `Test` and some `GetVerify` ?

Reply:
The Test functions execute transactions, they aren't in use yet.

Reply:
Would be great to execute transactions and verify them ina single go. Are you actively working on it or what is the plan?